ldap_auth_provider.py

```diff
@@ -84,6 +86,9 @@ def __init__(self, config, account_handler):
         self.ldap_active_directory = config.active_directory
         if self.ldap_active_directory:
             self.ldap_default_domain = config.default_domain
+            # This will be set to value of Active Directory root domain of type str
+            # or to None in case of error while retrieving Active Directory root domain
```

Comment:
and yet sometimes it is False?

```suggestion
            # Either: the Active Directory root domain (type str); None if there was
            # an error while retrieving Active Directory root domain; or False if ...
```

Reply:
I think that rather than having one of three possible types, it might be clearer to use:
 * `None`: not yet checked
 * empty string `""`: no root domain, or error
 * other string: the root domain.